filebeat/harvester/forwarder.go

```diff
@@ -0,0 +1,57 @@
+package harvester
+
+import (
+	"errors"
+
+	"github.com/elastic/beats/filebeat/util"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+type Outlet interface {
+	SetSignal(signal <-chan struct{})
+	OnEventSignal(data *util.Data) bool
+	OnEvent(data *util.Data) bool
+}
+
+type Forwarder struct {
+	Config     ForwarderConfig
+	Outlet     Outlet
+	Processors *processors.Processors
+}
+
+type ForwarderConfig struct {
+	common.EventMetadata `config:",inline"`      // Fields and tags to add to events.
+	InputType            string                  `config:"input_type"`
+	Pipeline             string                  `config:"pipeline"`
+	Module               string                  `config:"_module_name"`  // hidden option to set the module name
+	Fileset              string                  `config:"_fileset_name"` // hidden option to set the fileset name
+	Processors           processors.PluginConfig `config:"processors"`
+}
+
+// updateState updates the prospector state and forwards the event to the spooler
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported method Forwarder.Forward should be of the form "Forward ..."